### PR TITLE
Feature/add reducers and actions

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- [x] Make sure that there are no linter errors.
- [x] Make sure that you used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Make sure that you documented your work in a professional way.
### JavaScript requirements
- [x] Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
### Project requirements
- [x] Book Store: 
   - [x] Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
   - [x] Create a directory that will contain all your Redux logic (/src/redux)
   - [x] Configure a Redux store (/src/redux/store.js)
   - [x] Define a slice of state for books that:
        - [x] Includes an array of books (initial state: empty array)
       - [x] Includes a reducer that adds a book
      - [x] Includes a reducer that removes a book
  - [x] Define a slice of state for categories that:
      - [x] Includes an array of categories (initial state: empty array)
      - [x] Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
- [x] Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files.
   - [x] In the browser tested ; it render without problems, [this](https://benawi.github.io/Microverse-React-Bookstore/)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    case REMOVE_BOOK: {
+      const { bookID } = action;
+      return state.filter((book) => book.id !== bookID);
+    }
+    default:
+      return state;
+  }
+};
+
+export default reducer;
```
Comment (GzimAsani) [inline src/redux/books/booksSlice.js]: - [ ] You need to import { createSlice } from "@reduxjs/toolkit" and use it to create a reduxer. Refer to this [document](https://redux-toolkit.js.org/api/createslice). You did something similar to categories slice

Example:
```
import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

interface CounterState {
  value: number
}

const initialState = { value: 0 } as CounterState

const counterSlice = createSlice({
  name: 'counter',
  initialState,
  reducers: {
    increment(state) {
      state.value++
    },
    decrement(state) {
      state.value--
    },
    incrementByAmount(state, action: PayloadAction<number>) {
      state.value += action.payload
    },
  },
})

export const { increment, decrement, incrementByAmount } = counterSlice.actions
export default counterSlice.reducer
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Benawi                      :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with adding toolkit and react-redux on your package.json file  +1: 

 ## Required Changes 

- [ ] Check for the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
## Changes ♻
- [x] { createSlice } imported  from "@reduxjs/toolkit" and used to create a reduxer.

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      const bookIndex = state.findIndex((book) => book.id === action.payload);
+      if (bookIndex !== -1) {
+        state.splice(bookIndex, 1);
+      }
+    },
+  },
```
Comment (TedLivist) [inline src/redux/books/booksSlice.js:16]: - [ ] **[OPTIONAL]** It would be better to use `filter()` instead of `splice()` to avoid mutating the state.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Benawi,

✔️ Use of `createSlice` for actions

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**